### ISSUE TITLE
siteexport over ssl on IIS does not work

### ISSUE THREAD

**Issue by mihadoo** [i-net-software/dokuwiki-plugin-siteexport#33]
Hi, I have configured IIS to allow only SSL connections to dokuwiki. From that moment on siteexport stopped working. Firefox reports the following:

File not found
Firefox can't find the file at https://myserver.com/wiki/lib/exe/fetch.php?cache=nocache&siteexport=5b6e16b90a34536a886b1f21f214f1d5&sectok=f8b1e3b1a9b16f92ea773696aad7fa94&media=wiki:siteexport.zip.

The same works OK over http, but it does not work over https :(

**Comment by gamma** [comment]
Can you please send a log file with logging set to debug to tools[at]inetsoftware.de?

**Comment by mihadoo** [comment]
I am not sure if I have enabled the log file correctly. I have done this:

https://myserver.com/wiki/doku.php?id=sup&DebugFile=siteexport.log&DebugLevel=1&addParams=1&depth=0&depthType=1.0&do=siteexport&ens=sup&template=dokuwiki

After executing this no log file is created. Maybe I have to specify something else? The result that is displayed is still:

Firefox can't find the file at https://myserver.com/wiki/lib/exe/fetch.php?cache=nocache&siteexport=21a6ebd110d237c6309dd70bf5337276&sectok=5367ca9eabaaa8535d86172ac74e9a41&media=wiki:siteexport.zip

**Comment by gamma** [comment]
Can you please enable the debug options in the settings of DW? The debug file should be either relative to DW or better: an absolute path.

**Comment by gamma** [comment]
Ok, the file has not been created due to download errors. Whenever the site export tries to fetch a file using ssl it receives an HTTP status '-100' (never seen this).

You are also using authentication to get to your sites. Did you define this in your settings or do you export with an authenticated account? If you're using an authenticated account, can you please copy the download URL from the fields on the bottom of the site export page, log out and try to load the URL from the clipboard? Does this work?

What kind of authentication mechanism do you use?

**Comment by mihadoo** [comment]
Hi Gerry

Thank you for replying. I tried both: Clicking on the Start button and pasting the URL but the result is always the same. The log you received was by pasting the URL. I use standard dokuwiki authentication with username and password. Also by entering the username and password in dokuwiki settings the problem remains the same.

Looking at this https://social.msdn.microsoft.com/Forums/en-US/b36a682f-e7a4-497a-a588-4da24b6a77cb/https-post-returning-100-continue-makes-the-webclient-throw-protocol-violation-exception?forum=netfxnetcom I assume that the HTTPClient for SSL is not implemented correctly in the dokuwiki file wiki\inc\HTTPClient.php for SSL connections to IIS. 

Maybe I should report this to the dokuwiki bug list on https://github.com/splitbrain/dokuwiki/issues ?

Kind regards
Miha  Sokolov
----------------------------------------------- 
E-Mail: mihasokolov@hotmail.com 
----------------------------------------------- 


Date: Mon, 18 May 2015 02:34:20 -0700

**Comment by gamma** [comment]
Ok, yes, I agree. Please file a report at the dokuwiki issue tracker.

Would allowing non-ssl connections from your own server be an option as a workaround?

**Comment by mihadoo** [comment]
Hi Gerry

Currently I had to enable non-ssl connections for all clients so that my daily automated siteexport keeps working. The reason why I disabled non-ssl connections to the server is that for some years we used non-ssl ftp connections for our website upload and somebody obviously intercepted the password because one day our website was infected with virus files. Therefore I would like to use SSL connections only as soon as possible for all sites that require authentication.

The solution would of course be to allow non-ssl connections from localhost only, but I do not know how to configure this in IIS. I will google it...



Date: Mon, 18 May 2015 03:56:58 -0700
